cpp/ql/src/semmle/code/cpp/dataflow/internal/AddressFlow.qll

```diff
@@ -218,7 +247,9 @@ private predicate referenceToUpdate(Expr reference, Expr outer, ControlFlowNode
     not stdIdentityFunction(call.getTarget()) and
     not stdAddressOf(call.getTarget()) and
     exists(ReferenceType rt | rt = outer.getType().stripTopLevelSpecifiers() |
-      not rt.getBaseType().isConst()
+      not rt.getBaseType().isConst() or
+      rt.getBaseType().getUnspecifiedType() =
+        any(PointerWrapper wrapper | not wrapper.pointsToConst())
```

Comment:
In this new rule, `rt` is something like `const shared_ptr<int> &`: there are two levels of indirection before the `int`. Therefore I'd expect this rule to be in a new "reference-to-pointer to update" predicate.

Reply:
Fixed in 5aeaab7c6d7ff786c970b41f1bebce8c6040d838.